Add VE use case docs (#8933)
mendix/docs: content/en/docs/refguide/modeling/domain-model/entities/view-entities/multilingual-apps.md

---
title: "Multilingual Apps and Translations"
url: /refguide/multilingual-apps/
weight: 50
---

## Introduction

Use view entities to create apps that provide translations for your users.

## Use Case

> For this purpose of this use case, the following domain model is used:

Reviewer: ```suggestion
For a use case described below, the following domain model is used:
```

{{< figure src="/attachments/refguide/modeling/domain-model/view-entities/multilingual-apps/domain-model.png" >}}

The entity *ProductTranslation* contains a translation of a product’s description to a language indicated by *LanguageCode*. *LanguageCode* is a standard 5-letter language-country code (for example, `en_US`, `nl_NL`, `ko_KR`).

The descriptions in the default language (in this example, `en_US`) are stored in the *Product* entity, whereas translated descriptions (`nl_NL` and `ko_KR`) are stored in *ProductTranslation*. 

## Create an Overview Page

> To use a view entity that translate languages, ensure your app supports multiple languages. To do this, follow these steps:

Reviewer: ```suggestion
To use a view entity that translates languages, ensure your app supports multiple languages. Follow the steps below:
```

1. Open your **App Settings** and click the **Languages** tab.
2. Click **Add** and use the drop-down to select the language (or languages) you want to include in your app.
3. Click **OK** > **OK**.

Combine the languages in the two entities into a single view. To do this, join the *Product* and *ProductTranslation* entities by creating a new view entity.

1. Open your domain model and add a new view entity named *AllProductTranslationVE*.
2. Add the following query for your entity:

```
SELECT
    p.ProductId as ProductId,
    p.ProductName as ProductName,
    p.QuantityPerUnit as QuantityPerUnit,
    p.Discontinued as Discontinued,
    pt.LanguageCode as LanguageCode,
    coalesce(pt.Description, p.Description) as Description
FROM Shop.Product p
    JOIN p/Shop.ProductTranslation_Product/Shop.ProductTranslation pt
UNION
SELECT
    p.ProductId as ProductId,
    p.ProductName as ProductName,
    p.QuantityPerUnit as QuantityPerUnit,
    p.Discontinued as Discontinued,
    cast('en_US' as STRING) as LanguageCode,
    p.Description as Description
FROM Shop.Product p
```

> {{< figure src="/attachments/refguide/modeling/domain-model/view-entities/multilingual-apps/all-product-translation-ve.png" width="200" >}}

Reviewer: Please check indentation for this list
